data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -253,12 +261,144 @@ public Optional<Number> executeUpdate(PreparedQuery<?, Number> preparedQuery) {
 
     @Override
     public <T> int delete(DeleteOperation<T> operation) {
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(operation.getRootEntity());
+        T entity = operation.getEntity();
+        CosmosContainer container = getContainer(persistentEntity);
+        ObjectNode item = serializeToTree(entity, Argument.of(operation.getRootEntity()));
+        CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        CosmosItemResponse<?> cosmosItemResponse = container.deleteItem(item, options);
+        if (cosmosItemResponse.getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
+            return 1;
+        }
         return 0;
     }
 
     @Override
     public <T> Optional<Number> deleteAll(DeleteBatchOperation<T> operation) {
-        return Optional.empty();
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(operation.getRootEntity());
+        CosmosContainer container = getContainer(persistentEntity);
+        CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        Argument<T> arg = Argument.of(operation.getRootEntity());
+        Iterator<T> iterator = operation.iterator();
+        String partitionKeyDefinition = getPartitionKey(persistentEntity);
+        int deletedCount;
+        if (StringUtils.isEmpty(partitionKeyDefinition)) {
+            deletedCount = 0;
+            // Bulk operations delete won't work without partition key so deleting one by one document
+            while (iterator.hasNext()) {
```

Comment:
You can just write for `(T entity : operation) {}`, also the same a few lines down

Reply:
Changed